src/compiler/crystal/codegen/codegen.cr

```diff
@@ -2213,7 +2213,8 @@ module Crystal
     end
 
     private def c_memset_fun
-      name = @program.bits64? ? "llvm.memset.p0i8.i64" : "llvm.memset.p0i8.i32"
+      p0 = {% if LibLLVM::IS_LT_150 %} "p0i8" {% else %} "p0" {% end %}
+      name = "llvm.memset.#{p0}.#{@program.bits64? ? "i64" : "i32"}"
```

Comment:
This change now builds the string representing the function name at runtime. Did you consider keeping the full strings and just selecting between four options instead of two?